<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CollectionWidget</class>
 <widget class="QWidget" name="CollectionWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>256</width>
    <height>284</height>
   </rect>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <property name="margin">
    <number>0</number>
   </property>
   <item>
    <widget class="QToolBar" name="col_toolbar">
     <addaction name="actionGenreMode"/>
     <addaction name="actionArtistMode"/>
     <addaction name="actionAlbumMode"/>
     <addaction name="actionSongMode"/>
     <addaction name="actionSqlMode"/>
     <addaction name="separator"/>
     <addaction name="actionIconView"/>
     <addaction name="actionSetImage"/>
     <addaction name="actionGetImageFromLastFM"/>
    </widget>
   </item>
   <item>
    <widget class="QFrame" name="frame_2">
     <property name="frameShape">
      <enum>QFrame::NoFrame</enum>
     </property>
     <property name="lineWidth">
      <number>0</number>
     </property>
     <layout class="QHBoxLayout">
      <property name="spacing">
       <number>0</number>
      </property>
      <property name="margin">
       <number>0</number>
      </property>
      <item>
       <widget class="QLabel" name="subsetLabel">
        <property name="text">
         <string>TextLabel</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="subsetDisableButton">
        <property name="acceptDrops">
         <bool>false</bool>
        </property>
        <property name="toolTip">
         <string>Reset filter</string>
        </property>
        <property name="autoFillBackground">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>...</string>
        </property>
        <property name="icon">
         <iconset resource="../../src/resource.qrc">
          <normaloff>:/icons/clear_filter.png</normaloff>:/icons/clear_filter.png</iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>24</width>
          <height>24</height>
         </size>
        </property>
        <property name="autoRaise">
         <bool>true</bool>
        </property>
        <property name="arrowType">
         <enum>Qt::NoArrow</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="filterLineEdit">
        <property name="toolTip">
         <string>Enter search substring</string>
        </property>
        <property name="statusTip">
         <string>Enter search substring</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="CollectionView" name="colView">
     <property name="contextMenuPolicy">
      <enum>Qt::ActionsContextMenu</enum>
     </property>
     <property name="acceptDrops">
      <bool>true</bool>
     </property>
     <property name="toolTip">
      <string>Drag'n'Drop files to the collection</string>
     </property>
     <property name="editTriggers">
      <set>QAbstractItemView::EditKeyPressed|QAbstractItemView::SelectedClicked</set>
     </property>
     <property name="dragEnabled">
      <bool>true</bool>
     </property>
     <property name="flow">
      <enum>QListView::TopToBottom</enum>
     </property>
     <property name="isWrapping" stdset="0">
      <bool>true</bool>
     </property>
     <addaction name="actionAddToCollection"/>
     <addaction name="actionEdit"/>
     <addaction name="actionRemoveFromCollection"/>
     <addaction name="actionSetImage"/>
     <addaction name="actionGetImageFromLastFM"/>
    </widget>
   </item>
  </layout>
  <action name="actionGenreMode">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../../src/resource.qrc">
     <normaloff>:/icons/def_genre.png</normaloff>:/icons/def_genre.png</iconset>
   </property>
   <property name="text">
    <string>Genre</string>
   </property>
   <property name="statusTip">
    <string>Show genres</string>
   </property>
  </action>
  <action name="actionArtistMode">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../../src/resource.qrc">
     <normaloff>:/icons/def_artist.png</normaloff>:/icons/def_artist.png</iconset>
   </property>
   <property name="text">
    <string>Artist</string>
   </property>
   <property name="statusTip">
    <string>Show artists</string>
   </property>
  </action>
  <action name="actionAlbumMode">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../../src/resource.qrc">
     <normaloff>:/icons/def_album.png</normaloff>:/icons/def_album.png</iconset>
   </property>
   <property name="text">
    <string>Album</string>
   </property>
   <property name="statusTip">
    <string>Show albums</string>
   </property>
  </action>
  <action name="actionSongMode">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../../src/resource.qrc">
     <normaloff>:/icons/def_song.png</normaloff>:/icons/def_song.png</iconset>
   </property>
   <property name="text">
    <string>Song</string>
   </property>
   <property name="statusTip">
    <string>Show songs</string>
   </property>
  </action>
  <action name="actionIconView">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../../src/resource.qrc">
     <normaloff>:/icons/iconview.png</normaloff>:/icons/iconview.png</iconset>
   </property>
   <property name="text">
    <string>Icon View</string>
   </property>
   <property name="statusTip">
    <string>View icons for collection items</string>
   </property>
  </action>
  <action name="actionSetImage">
   <property name="icon">
    <iconset resource="../../src/resource.qrc">
     <normaloff>:/icons/col_art.png</normaloff>:/icons/col_art.png</iconset>
   </property>
   <property name="text">
    <string>Set Image</string>
   </property>
   <property name="statusTip">
    <string>Set image for collection item</string>
   </property>
  </action>
  <action name="actionGetImageFromLastFM">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="../../src/resource.qrc">
     <normaloff>:/icons/download_art.png</normaloff>:/icons/download_art.png</iconset>
   </property>
   <property name="text">
    <string>Load image from Last.FM</string>
   </property>
   <property name="statusTip">
    <string>Load image from Last.FM</string>
   </property>
  </action>
  <action name="actionAddToCollection">
   <property name="icon">
    <iconset resource="../../src/resource.qrc">
     <normaloff>:/icons/col_add.png</normaloff>:/icons/col_add.png</iconset>
   </property>
   <property name="text">
    <string>Add Item</string>
   </property>
   <property name="statusTip">
    <string>Create new collection item</string>
   </property>
  </action>
  <action name="actionRemoveFromCollection">
   <property name="icon">
    <iconset resource="../../src/resource.qrc">
     <normaloff>:/icons/col_rem.png</normaloff>:/icons/col_rem.png</iconset>
   </property>
   <property name="text">
    <string>Remove Item</string>
   </property>
   <property name="statusTip">
    <string>Remove selected item from collection</string>
   </property>
  </action>
  <action name="actionSqlMode">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../../src/resource.qrc">
     <normaloff>:/icons/def_sqllist.png</normaloff>:/icons/def_sqllist.png</iconset>
   </property>
   <property name="text">
    <string>SQL Playlists</string>
   </property>
   <property name="toolTip">
    <string>SQL Playlists</string>
   </property>
  </action>
  <action name="actionEdit">
   <property name="icon">
    <iconset resource="../../src/resource.qrc">
     <normaloff>:/icons/def_list.png</normaloff>:/icons/def_list.png</iconset>
   </property>
   <property name="text">
    <string>Edit</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>CollectionView</class>
   <extends>QListView</extends>
   <header>collectionview.h</header>
   <slots>
    <slot>sqlMode()</slot>
    <slot>editItem()</slot>
    <slot>filter(QString)</slot>
    <slot>slowFilter()</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../../src/resource.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionArtistMode</sender>
   <signal>triggered()</signal>
   <receiver>colView</receiver>
   <slot>artistMode()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>268</x>
     <y>219</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionRemoveFromCollection</sender>
   <signal>triggered()</signal>
   <receiver>colView</receiver>
   <slot>removeItem()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>140</x>
     <y>216</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>colView</sender>
   <signal>setVisibleSubsetWidgets(bool)</signal>
   <receiver>subsetLabel</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>136</x>
     <y>245</y>
    </hint>
    <hint type="destinationlabel">
     <x>35</x>
     <y>146</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionAddToCollection</sender>
   <signal>triggered()</signal>
   <receiver>colView</receiver>
   <slot>addItem()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>140</x>
     <y>216</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>subsetDisableButton</sender>
   <signal>clicked()</signal>
   <receiver>colView</receiver>
   <slot>clearSubset()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>80</x>
     <y>146</y>
    </hint>
    <hint type="destinationlabel">
     <x>136</x>
     <y>245</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSongMode</sender>
   <signal>triggered()</signal>
   <receiver>colView</receiver>
   <slot>songMode()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>208</x>
     <y>219</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionGenreMode</sender>
   <signal>triggered()</signal>
   <receiver>colView</receiver>
   <slot>genreMode()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>268</x>
     <y>219</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionAlbumMode</sender>
   <signal>triggered()</signal>
   <receiver>colView</receiver>
   <slot>albumMode()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>268</x>
     <y>219</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSetImage</sender>
   <signal>triggered()</signal>
   <receiver>colView</receiver>
   <slot>setImage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>136</x>
     <y>253</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>colView</sender>
   <signal>setSubsetLabel(QString)</signal>
   <receiver>subsetLabel</receiver>
   <slot>setText(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>131</x>
     <y>173</y>
    </hint>
    <hint type="destinationlabel">
     <x>39</x>
     <y>128</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>colView</sender>
   <signal>setVisibleSubsetWidgets(bool)</signal>
   <receiver>subsetDisableButton</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>136</x>
     <y>245</y>
    </hint>
    <hint type="destinationlabel">
     <x>80</x>
     <y>146</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionGetImageFromLastFM</sender>
   <signal>triggered()</signal>
   <receiver>colView</receiver>
   <slot>loadImage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>141</x>
     <y>156</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionIconView</sender>
   <signal>triggered(bool)</signal>
   <receiver>colView</receiver>
   <slot>iconView(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>149</x>
     <y>151</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSqlMode</sender>
   <signal>triggered()</signal>
   <receiver>colView</receiver>
   <slot>sqlMode()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>127</x>
     <y>175</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionEdit</sender>
   <signal>triggered()</signal>
   <receiver>colView</receiver>
   <slot>editItem()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>127</x>
     <y>175</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>filterLineEdit</sender>
   <signal>textChanged(QString)</signal>
   <receiver>colView</receiver>
   <slot>filter(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>166</x>
     <y>54</y>
    </hint>
    <hint type="destinationlabel">
     <x>127</x>
     <y>179</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>filterLineEdit</sender>
   <signal>editingFinished()</signal>
   <receiver>colView</receiver>
   <slot>slowFilter()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>172</x>
     <y>63</y>
    </hint>
    <hint type="destinationlabel">
     <x>127</x>
     <y>187</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
